docs/react-v9/contributing/rfcs/shared/build-system/stop-styles-transforms.md

```diff
@@ -0,0 +1,162 @@
+
+[@layershifter](https://github.com/layershifter)
+
+## Background
+
+We've begun noticing a pattern where apps are bundled into multiple separate bundles. For example, there's the main bundle for the app itself and a separate bundle for a first UI piece fetched from a CDN. This problem is described in the following issues:
+
+- [microsoft/griffel#453](https://github.com/microsoft/griffel/issues/453) (_check this issue for more details_)
+- [microsoft/griffel#526](https://github.com/microsoft/griffel/issues/526)
+
+_TL;DR_: Griffel relies on the order of CSS classes, which can lead to clashes when multiple bundles are involved. See the simple example below:
+
+```html
+<!-- CURRENT STATE -->
+
+<!-- main bundle -->
+<style>
+  .order0 {
+    padding: 10px;
+  }
+  .order1 {
+    padding-left: 5px;
+  }
+</style>
+
+<!-- CDN bundle -->
+<style>
+  .order0 {
+    padding: 10px;
+  }
+</style>
+
+<!-- HTML -->
+
+<!--
+  🔴 We expect that a "div" below will have "padding-left: 5px", but instead, it has "padding: 10px".
+
+     The issue appears because the CDN bundle loads after the main bundle, resulting in style overrides. This occurs because ".order0" appears in both bundles, and CSS prioritizes the order of appearance. 
+  -->
+
+<div class="order0 order1"></div>
+```
+
+The single reliable solution is to prefix styles with a unique identifier. This prevents CSS rules from colliding, for example:
+
+```html
+<!-- PROPOSAL -->
+
+<!-- main bundle -->
+<style>
+  .main-order0 {
+    padding: 10px;
+  }
+  .main-order1 {
+    padding-left: 5px;
+  }
+</style>
+
+<!-- CDN bundle -->
+<style>
+  .order0 {
+    padding: 10px;
+  }
+</style>
+
+<!-- HTML -->
+
+<!--
+  ✅ Now the "div" below will have "padding-left: 5px" as CSS rules don't clash anymore.
+  -->
+
+<div class="main-order0 main-order1"></div>
+```
+
+## Problem statement
+
+Integrating prefixing into Griffel is relatively straightforward. However, we currently undergo a pre-processing step in `@fluentui/react-components` that modifies JavaScript code.
+
+<details>
+  <summary>Original code</summary>
+
+```ts
+// packages/react-components/react-menu/src/components/MenuDivider/useMenuDividerStyles.styles.ts
+// 📝 output is simpfied
+
+const useStyles = makeStyles({
+  root: {
+    ...shorthands.margin('4px', '-5px', '4px', '-5px'),
+  },
+});
+```
+
+</details>
+
+<details>
+  <summary>Pre-processed code</summary>
+
+```ts
+// @fluentui/react-menu/lib/components/MenuDivider/useMenuDividerStyles.styles.js
+// 📝 output is simpfied
+
+const useStyles = /*#__PURE__*/ __styles(
+  {
+    root: {
+      B6of3ja: 'fvjh0tl',
+      t21cq0: ['f1rnx978', 'f1q7jvqi'],
+    },
+  },
+  {
+    d: ['.fvjh0tl{margin-top:4px;}', '.f1rnx978{margin-right:-5px;}', '.f1q7jvqi{margin-left:-5px;}'],
+  },
+);
+```
+
+</details>
+
+Addressing prefixing for pre-processed styles presents a challenge, as they have already been transformed into CSS rules and classes.
```

Comment:
I have no deep knowledge about griffel implementation but can't we emit some kind o metadata to these post-processed outputs that griffel webpack(use your bundler here) loader would understand and would mitigate this issues?

**Contrived example:**

```diff
+ const useStyles = /*#__PURE__*/ __stylesFactory('unique-prefix')(
- const useStyles = /*#__PURE__*/ __styles(
  {
    root: {
      B6of3ja: 'fvjh0tl',
      t21cq0: ['f1rnx978', 'f1q7jvqi'],
    },
  },
  {
    d: ['.fvjh0tl{margin-top:4px;}', '.f1rnx978{margin-right:-5px;}', '.f1q7jvqi{margin-left:-5px;}'],
  },
);
```

Reply:
I think that it missed this in RFC as was focused more on CSS output, but as Griffel has also runtime version (and AOT works with that) and it has limitations.

For that suggestion, we will need:
- **Build time**
  - Process prefixing during both stages of AOT and CSS (_with current proposals it's limited only to AOT_)
  - 🔼 That approach is against the upcoming performance improvements like merging to two loaders into a single one
- **Runtime**
  - Generate prefixed CSS with on `makeStyles()` calls
  - Significantly complicate at least `__styles()`: the function will have to handle prefixing and process both mappings and CSS rules. That will regress performance of it and is against an idea of AOT.

/cc @spmonahan

Reply:
Added this to "Current workaround" section.